docs/source/api/data-management/graphs.rst

```diff
@@ -0,0 +1,260 @@
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. highlight:: cpp
+
+.. _api_graphs:
+
+======
+Graphs
+======
+
+Refer to :ref:`Developer Guide: Graphs <dm_graphs>`.
+
+.. _graph_programming_interface:
+
+---------------------
+Programming interface
+---------------------
+
+All types and functions in this section are declared in the
+``oneapi::dal::preview`` namespace and are available via inclusion of the
+``oneapi/dal/graph/common.hpp`` header file.
+
+Graph
+-----
+
+The :txtref:`graph` concept is represented by the types with the ``_graph`` suffix and all of them are
+:term:`reference-counted <Reference-counted object>`:
+
+1. The instance stores pointers to the graph :capterm:`topology`
+   and :capterm:`attributes <Attribute>` of vertices and edges.
+
+2. The reference count indicating how many graph objects refer to the same implementation.
+
+3. The graph increments the reference count
+   for it to be equal to the number of graph objects sharing the same implementation.
+
+4. The graph decrements the reference count when the
+   graph goes out of the scope. If the reference count is zero, the graph
+   frees its implementation.
+
+.. _api_graph_template_type:
+
+The :txtref:`graph` types are defined as templated classes:
+
+.. code-block:: cpp
+
+  template <typename VertexValue,
+            typename EdgeValue,
+            typename GraphValue,
+            typename IndexType,
+            typename Allocator>
+  class [graph_name]_graph;
+
+.. .. namespace:: oneapi::dal::preview
+.. .. struct:: graph_traits
+
+.. list-table:: 
+   :widths: 20 40 40 
+   :header-rows: 1
+
+   * - Type name
+     - Description
+     - Supported types
+   * - ``VertexValue``
+     - Type of the vertex :capterm:`attribute <Attribute>` values
+     - :ref:`Empty value <api_empty_value>`
+   * - ``EdgeValue``
+     - Type of the edge :capterm:`attribute <Attribute>` values
+     - ``std::int32``, ``double``, :ref:`Empty value <api_empty_value>`  
+   * - ``GraphValue``
+     - Type of the graph :capterm:`attribute <Attribute>` value
+     - :ref:`Empty value <api_empty_value>`  
+   * - ``IndexType``
+     - Type of the :capterm:`vertex indices <Vertex index>`
+     - ``std::int32`` 
+   * - ``Allocator``
+     - graph allocator type.
+     - C++17 (ISO/IEC 14882:2017) compliant allocator
+
+
+:ref:`Empty value <api_empty_value>` tag structure is used to define 
+the absence of specified attribute of graph.
+
+.. _api_empty_value:
+
+.. onedal_class:: oneapi::dal::preview::empty_value
+
+Graph class contains default and move constructor, assignment.
```

Comment:
```suggestion
Graph class contains the default and the move constructor as well as the move assignment function.
```

`assignment` -- maybe `move assignment operator`. `move assignment function` or something like this? currently this is mot grammatically correct